db/hints/manager.hh

```diff
@@ -104,12 +104,14 @@ public:
                 send_state::segment_replay_failed>>;
 
             struct send_one_file_ctx {
-                send_one_file_ctx(std::unordered_map<table_schema_version, column_mapping>& last_schema_ver_to_column_mapping)
+                send_one_file_ctx(std::unordered_map<table_schema_version, column_mapping>& last_schema_ver_to_column_mapping, db::replay_position last_sent_rp)
                     : schema_ver_to_column_mapping(last_schema_ver_to_column_mapping)
+                    , last_sent_rp(last_sent_rp)
```

Comment:
This is very confusing.
There is not point in trying to save on std::optional here out of a sudden.

There are three distinctive things here:
 1) The position from which we need to restart sending: `_last_not_complete_rp`.
 2) Last attempted send position: `last_sent_rp` (should be `last_attempted_rp`).
 3) First failed position: `first_failed_rp`.

Last two should be optional since you may end up not setting them at all.
And they influence on the value of `_last_not_complete_rp` on the exit from `send_one_file()`.

Reply:
Yeah, it makes sense to do it in this way. I changed `last_attempted_rp` to be an `std::optional`